// @progress/kendo-vue-layout/dist/npm/menu/components/MenuItemArrow.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MenuItemArrowVue2 = exports.MenuItemArrow = exports.leftArrowClass = exports.rightArrowClass = exports.downArrowClass = void 0; // @ts-ignore

var Vue = require("vue");

var allVue = Vue;
var gh = allVue.h;

var misc_1 = require("../utils/misc");
/**
 * @hidden
 */


exports.downArrowClass = 'k-i-arrow-60-down';
/**
 * @hidden
 */

exports.rightArrowClass = 'k-i-arrow-60-right';
/**
 * @hidden
 */

exports.leftArrowClass = 'k-i-arrow-60-left';
/**
 * @hidden
 */

var MenuItemArrowVue2 = {
  name: 'KendoMenuItemArrow',
  props: {
    itemId: String,
    dir: String,
    verticalMenu: Boolean
  },
  // @ts-ignore
  setup: !gh ? undefined : function () {
    var v3 = !!gh;
    return {
      v3: v3
    };
  },
  render: function render(createElement) {
    var h = gh || createElement;
    return h("span", {
      "class": "k-menu-expand-arrow"
    }, [h("span", {
      "class": this.getArrowClassName(),
      role: "presentation",
      attrs: this.v3 ? undefined : {
        role: "presentation"
      }
    })]);
  },
  methods: {
    getArrowClassName: function getArrowClassName() {
      var _a;

      var childrenPosition = misc_1.getChildrenPosition(this.$props.itemId, this.$props.verticalMenu === true, this.$props.dir === 'rtl');
      return _a = {
        'k-icon': true
      }, _a[exports.downArrowClass] = childrenPosition === 'downward', _a[exports.rightArrowClass] = childrenPosition === 'rightward', _a[exports.leftArrowClass] = childrenPosition === 'leftward', _a;
    }
  }
};
exports.MenuItemArrowVue2 = MenuItemArrowVue2;
/**
 * @hidden
 */

var MenuItemArrow = MenuItemArrowVue2;
exports.MenuItemArrow = MenuItemArrow;
