// @progress/kendo-vue-inputs/dist/es/colors/HexInput.js
// @ts-ignore
import * as Vue from 'vue';
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version[0] === '3';
import { parseColor } from './utils/color-parser';
import { isPresent } from './utils/misc';
import { Input } from '../input/Input';
/**
 * @hidden
 */

var HexInputVue2 = {
  name: 'KendoHexInput',
  props: {
    hex: String,
    disabled: Boolean
  },
  data: function data() {
    return {
      hex: this.$props.hex,
      originalHex: this.$props.hex
    };
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  // @ts-ignore
  render: function render(createElement) {
    var h = gh || createElement;
    return h(Input, {
      value: this.hex,
      attrs: this.v3 ? undefined : {
        value: this.hex,
        disabled: this.$props.disabled
      },
      onChange: this.onChange,
      on: this.v3 ? undefined : {
        "change": this.onChange,
        "blur": this.onBlur
      },
      onBlur: this.onBlur,
      disabled: this.$props.disabled
    });
  },
  methods: {
    onChange: function onChange(event) {
      var hex = event.target.value;
      var value = parseColor(hex, 'rgba');
      this.hex = hex;

      if (isPresent(value)) {
        this.$props.onHexChange(hex, value, event);
      }
    },
    onBlur: function onBlur() {
      if (!isPresent(parseColor(this.hex, 'rgba'))) {
        this.hex = this.originalHex;
      }
    }
  }
};
/**
 * @hidden
 */

var HexInput = HexInputVue2;
export { HexInput, HexInputVue2 };
