// @progress/kendo-vue-layout/dist/npm/tabstrip/TabStripTab.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TabStripTabVue2 = exports.TabStripTab = void 0; // @ts-ignore

var Vue = require("vue");

var allVue = Vue;
var gh = allVue.h;
var ref = allVue.ref;
var inject = allVue.inject;

var kendo_vue_common_1 = require("@progress/kendo-vue-common");
/**
 * @hidden
 */


var TabStripTabVue2 = {
  name: 'KendoTabStripTab',
  props: {
    disabled: Boolean,
    contentClassName: String,
    title: String,
    titleRender: [String, Function, Object]
  },
  inject: {
    addRenderTitle: {
      default: null
    },
    addTab: {
      default: null
    },
    removeTab: {
      default: null
    }
  },
  created: function created() {
    this.tabId = kendo_vue_common_1.guid();
    this.addTab({
      title: this.$props.title,
      id: this.tabId,
      disabled: this.$props.disabled,
      contentClassName: this.$props.contentClassName
    });
  },
  destroyed: !!gh ? undefined : function () {
    this.removeTab(this.tabId);
  },
  // @ts-ignore
  unmounted: function unmounted() {
    this.removeTab(this.tabId);
  },
  // @ts-ignore
  setup: !gh ? undefined : function () {
    var v3 = !!gh;
    return {
      v3: v3
    };
  },
  render: function render(createElement) {
    var h = gh || createElement;
    var titleRender = this.$props.titleRender ? kendo_vue_common_1.templateRendering.call(this, this.$props.titleRender, kendo_vue_common_1.getListeners.call(this)) : null;

    if (titleRender) {
      this.addRenderTitle(this.tabId, titleRender);
    }

    return h("div", [kendo_vue_common_1.getDefaultSlots(this)]);
  }
};
exports.TabStripTabVue2 = TabStripTabVue2;
/**
 * @hidden
 */

var TabStripTab = TabStripTabVue2;
exports.TabStripTab = TabStripTab;
