// @progress/kendo-vue-form/dist/npm/FormElement.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FormElementVue2 = exports.FormElement = void 0; // @ts-ignore

var Vue = require("vue");

var allVue = Vue;
var gh = allVue.h;

var package_metadata_1 = require("./package-metadata");

var kendo_vue_common_1 = require("@progress/kendo-vue-common");
/**
 * @hidden
 */


var FormElementVue2 = {
  name: 'KendoFormElement',
  props: {
    horizontal: Boolean
  },
  created: function created() {
    kendo_vue_common_1.validatePackage(package_metadata_1.packageMetadata);
  },
  inject: {
    kendoForm: {
      default: null
    }
  },
  computed: {
    formElementClassName: function formElementClassName() {
      return {
        'k-form': true,
        'k-form-horizontal': this.$props.horizontal === true
      };
    }
  },
  methods: {
    handleSubmit: function handleSubmit(e) {
      if (this.kendoForm) {
        this.kendoForm.onSubmit(e);
      }
    }
  },
  // @ts-ignore
  setup: !gh ? undefined : function () {
    var v3 = !!gh;
    return {
      v3: v3
    };
  },
  render: function render(createElement) {
    var h = gh || createElement;
    var defaultSlots = kendo_vue_common_1.getDefaultSlots(this);
    return h("form", {
      "class": this.formElementClassName,
      onSubmit: this.handleSubmit,
      on: this.v3 ? undefined : {
        "submit": this.handleSubmit
      }
    }, [defaultSlots]);
  }
};
exports.FormElementVue2 = FormElementVue2;
/**
 * @hidden
 */

var FormElement = FormElementVue2;
exports.FormElement = FormElement;
