// @progress/kendo-vue-editor/dist/npm/dialogs/viewHtml.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ViewHtmlDialogVue2 = exports.ViewHtmlDialog = void 0; // @ts-ignore

var Vue = require("vue");

var allVue = Vue;
var gh = allVue.h;

var kendo_vue_dialogs_1 = require("@progress/kendo-vue-dialogs");

var kendo_vue_buttons_1 = require("@progress/kendo-vue-buttons");

var kendo_editor_common_1 = require("@progress/kendo-editor-common");

var kendo_vue_intl_1 = require("@progress/kendo-vue-intl");

var messages_1 = require("./../messages");

var props_key_1 = require("./../utils/props-key");
/**
 * @hidden
 */


var ViewHtmlDialogVue2 = {
  name: 'KendoViewHtmlDialog',
  props: {
    view: Object,
    settings: Object,
    dir: String
  },
  inject: {
    kendoLocalizationService: {
      default: null
    }
  },
  mounted: function mounted() {
    this.htmlArea = this.v3 ? this.htmlAreaRef : this.$refs.htmlArea;
  },
  // @ts-ignore
  setup: !gh ? undefined : function () {
    var v3 = !!gh;
    return {
      v3: v3
    };
  },
  // @ts-ignore
  render: function render(createElement) {
    var _this2 = this;

    var _this = this;

    var h = gh || createElement;
    var _a = this.$props,
        view = _a.view,
        settings = _a.settings,
        dir = _a.dir;
    var localization = kendo_vue_intl_1.provideLocalizationService(this);
    var _b = settings.messages,
        viewHtmlDialogTitle = _b.viewHtmlDialogTitle,
        viewHtmlCancel = _b.viewHtmlCancel,
        viewHtmlUpdate = _b.viewHtmlUpdate;
    var content = h("textarea", {
      "class": "k-textarea k-editor-textarea",
      style: {
        height: '280px',
        width: '490px'
      },
      ref: this.v3 ? function (el) {
        _this.htmlAreaRef = el;
      } : 'htmlArea',
      value: this.v3 ? kendo_editor_common_1.indentHtml(kendo_editor_common_1.getHtml(view.state)) : null,
      domProps: this.v3 ? undefined : {
        "value": kendo_editor_common_1.indentHtml(kendo_editor_common_1.getHtml(view.state))
      }
    });
    var actionButtons = [// @ts-ignore function children
    h(kendo_vue_buttons_1.Button, {
      type: "button",
      attrs: this.v3 ? undefined : {
        type: "button",
        dir: dir
      },
      onClick: this.onClose,
      on: this.v3 ? undefined : {
        "click": this.onClose
      },
      dir: dir,
      key: "cancel"
    }, this.v3 ? function () {
      return [localization.toLanguageString(viewHtmlCancel, messages_1.messages[viewHtmlCancel])];
    } : [localization.toLanguageString(viewHtmlCancel, messages_1.messages[viewHtmlCancel])]), // @ts-ignore function children
    h(kendo_vue_buttons_1.Button, {
      type: "button",
      attrs: this.v3 ? undefined : {
        type: "button",
        themeColor: 'primary',
        dir: dir
      },
      onClick: this.onUpdate,
      on: this.v3 ? undefined : {
        "click": this.onUpdate
      },
      themeColor: 'primary',
      dir: dir,
      key: "update"
    }, this.v3 ? function () {
      return [localization.toLanguageString(viewHtmlUpdate, messages_1.messages[viewHtmlUpdate])];
    } : [localization.toLanguageString(viewHtmlUpdate, messages_1.messages[viewHtmlUpdate])])];
    var dialog = // @ts-ignore function children
    h(kendo_vue_dialogs_1.Dialog, {
      title: localization.toLanguageString(viewHtmlDialogTitle, messages_1.messages[viewHtmlDialogTitle]),
      attrs: this.v3 ? undefined : {
        title: localization.toLanguageString(viewHtmlDialogTitle, messages_1.messages[viewHtmlDialogTitle]),
        closeIcon: true,
        dir: dir
      },
      onClose: this.onClose,
      on: this.v3 ? undefined : {
        "close": this.onClose
      },
      key: "dialog",
      closeIcon: true,
      dir: dir,
      style: dir === 'rtl' ? {
        textAlign: 'right'
      } : undefined
    }, this.v3 ? function () {
      return [content, // @ts-ignore function children
      h(kendo_vue_dialogs_1.DialogActionsBar, _this2.v3 ? function () {
        return [actionButtons];
      } : [actionButtons])];
    } : [content, h(kendo_vue_dialogs_1.DialogActionsBar, _this2.v3 ? function () {
      return [actionButtons];
    } : [actionButtons])]);
    return dialog;
  },
  methods: {
    onUpdate: function onUpdate() {
      var _a = this.$props,
          view = _a.view,
          settings = _a.settings;
      var html = kendo_editor_common_1.trimWhitespace(this.htmlArea ? this.htmlArea.value : '');
      var preserveWhitespace = props_key_1.editorPropsKey.getState(view.state).preserveWhitespace;
      kendo_editor_common_1.setHtml(html, settings.commandName, {
        preserveWhitespace: preserveWhitespace
      })(view.state, view.dispatch);
      this.onClose();
    },
    onClose: function onClose() {
      this.$props.view.focus();
      this.$emit('close');
    }
  }
};
exports.ViewHtmlDialogVue2 = ViewHtmlDialogVue2;
/**
 * @hidden
 */

var ViewHtmlDialog = ViewHtmlDialogVue2;
exports.ViewHtmlDialog = ViewHtmlDialog;
